src/components/Image/index.js

```diff
@@ -7,67 +7,58 @@ import ONYXKEYS from '../../ONYXKEYS';
 import {defaultProps, imagePropTypes} from './imagePropTypes';
 import RESIZE_MODES from './resizeModes';
 
-class Image extends React.Component {
-    componentDidMount() {
-        this.configureOnLoad();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.source === this.props.source) {
```

Comment:
Also i think we missing this logic - in useEffect we will have more items in deps - and it may cause issues when any prop like onLoad will changed but source will still the same. I will suggest to wrap function in React.memo and using second parameter to check if source is changed